### PR TITLE
Refactor /config for API 1.0 stability

### PR DESCRIPTION
**Pull request by tgraf**
*Note: This PR is a diff to https://github.com/cilium/cilium/pull/3389. Please don't merge.
See https://github.com/cilium/cilium/issues/3280 for background

**Summary of changes**:
- Refactor the daemon config to be declarative.

**How to test (optional)**:
all tests should pass.

Edit Thomas:
Fork of PR #3400 to rebase

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

File: api/v1/openapi.yaml [hunk 1]
```diff
@@ -1235,41 +1235,55 @@ definitions:
...
+  DaemonConfiguration:
+    description: |
+      Response to a daemon configuration request.
+    type: object
+    properties:
+      spec:
+        description: Changeable configuration
+        "$ref": "#/definitions/DaemonConfigurationSpec"
+      status:
+        description: Changeable configuration
```
Comment (rlenglet) [inline api/v1/openapi.yaml]: This is inconsistent with the description of the `DaemonConfigurationStatus` type below, which states that this contains also immutable configuration settings.

File: api/v1/openapi.yaml [hunk 2]
```diff
@@ -1235,41 +1235,55 @@ definitions:
...
+        "$ref": "#/definitions/DaemonConfigurationStatus"
+  DaemonConfigurationSpec:
+    description: |
+      General purpose structure to hold the controllable configuration of the daemon.
```
Comment (rlenglet) [inline api/v1/openapi.yaml]:
```
The controllable configuration of the daemon.
```

File: api/v1/openapi.yaml [hunk 3]
```diff
@@ -1235,41 +1235,55 @@ definitions:
...
+    type: object
+    properties:
+      options:
+        description: Changeable configuration
+        "$ref": "#/definitions/ConfigurationMap"
+      policy-enforcement:
+        description: "The policy-enforcement mode: default, always, never"
```
Comment (rlenglet) [inline api/v1/openapi.yaml]: Add a constraint to verify that the input values are in that enum:
```
pattern: "^default|always|never$"
```

**Review by rlenglet — APPROVED** [review]
Just small nitpicks. Nothing blocking.

**Comment by raybejjani** [comment]
@tgraf I didn't want to clobber your commits here since they passed tests. My updated branch is at raybejjani:api-cfg-corrected (https://github.com/cilium/cilium/pull/3457)
I've addressed the review comments and fixed two things:
- I removed a superfluous DaemonConfigurationStatus.PolicyEnforcement I left in place. The correct location is DaemonConfigurationStatus.Realized.PolicyEnforcement (which was already there).
- SSHMeta.SetPolicyEnforcement (test/helpers/cilium.go:399) was checking for policy-enforcement under .options. I had to hardcode in the appropriate field name since the constant it uses is not the json-name. The comment here suggests such an error would not pass tests but it did, so it might not be important.

**Comment by raybejjani** [comment]
This was tested in https://github.com/cilium/cilium/pull/3457

**Pull request merged**